client/src/App.js

```diff
@@ -8,101 +10,28 @@ function App() {
   const handleSubmit = (ev) => {
```

Comment:
Advanced tip: if you wanted this to be more reusable, you could convert it into a custom hook that looks something like this:

```js
useWeather({ city }) {
  const [result, setResult] = useState(null);

  useEffect(() => {
    if (!city) { return; }
    const controller = new AbortController()
    const params = new URLSearchParams({ cityName: city });
    fetch(`/weather?${params}`, { signal: controller.signal })
      .then((res) => res.json())
      .then((data) => setResult(data));
    return () => {
      controller.abort();
    };
  }, [city]);

  return result;
}
```

Then in this component, you'd use it like this:

```js
const [city, setCity] = useState(null)
const result = useWeather({ city })
```